net/suricata/files/suricata.init

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh /etc/rc.common
+# (C) 2018 Ashkan Jazayeri <ashkan@jazayeri.net>
+
+START=75
+USE_PROCD=1
+
+validate_suricata_section() {
+	uci_validate_section suricata suricata "${1}" \
+		'mode:string:af-packet' \
+		'queue:list(range(0,65535))' \
+		'verbose:range(0,4):0'
+}
+
+start_service() {
+	local mode number queue verbose
+	validate_suricata_section instance || {
+		echo "validation failed"
+		return 1
+	}
+	[ -d /var/log/suricata ] || mkdir /var/log/suricata
+	[ -f /var/run/suricata.pid -a -z $(pgrep suricata) ] && rm /var/run/suricata.pid && logger -t suricata[init_script] -p daemon.alert -s "Suricata was not closed properly or it has crashed. Successfully removed the previous /var/run/suricata.pid"
+	procd_open_instance
+	procd_set_param command /usr/bin/suricata
+    [ "$verbose" -gt 0 ] && {
+		procd_append_param command -$(printf 'v%.0s' $(seq 1 $verbose))
+		procd_set_param stdout 1
+		procd_set_param stderr 1
+	}
+	case "$mode" in
+		"af-packet" ) procd_append_param command --af-packet
+		;;
+		"nfq" )
+			[ -n "$queue" ] || {
+				logger -t suricata[init_script] -p daemon.emerg -s "No queue list provided. In NFQUEUE mode, a queue list must be specified under suricata config section (e.g. uci add_list suricata.instance.queue=9)"
+				return 1
+			}
+			for number in $queue; do procd_append_param command -q $number ;done
+		;;
+	esac
+	procd_append_param command --pidfile /var/run/suricata.pid
```

Comment:
This pidfile seems to only exist for your warning text.  You don't use it for config reloading, nor do you use procds signalling methods either. There's ```procd_set_param pidfile``` if you need the pidfile for something external, and 
```procd_send_signal``` for signalling you rprocess. see: https://openwrt.org/docs/guide-developer/procd-init-scripts

Reply:
This is a program specific arg:
> Write the process ID to file. Overrides the pid-file option in the configuration file and forces the file to be written when not running as a daemon.
http://suricata.readthedocs.io/en/latest/command-line-options.html#cmdoption-pidfile

As the program won't start if the pid file is already there from a previous crash (pidfile gets deleted on normal exits), I have to check for its existence and delete it beforehand. 
for that I think the user deserves a big fat warning.

also procd_send_signal won't do the job as it only fires "once" after each command arg change.
one needs to reload it each time a rule gets added/updated regardless of command arg changes.

Reply:
where do "rules" come from? if they're in files, mark the file properly with ```procd_set_param file```.  Can you just not provide the pidfile option at all?

Reply:
>where do "rules" come from?

already there ones: 
https://github.com/OISF/suricata/blob/master/rules/Makefile.am

rest of them: 
https://raw.githubusercontent.com/jasonish/suricata-intel-index/master/index.yaml

Enabling Emerging Threats free rules alone would add around 50 rule files to a basic setup which I think is not reasonable to track with procd. 
not to mention the list would not be consistent around different users with different subscriptions, so we have to add another script to handle the rule file listings and adding them for tracking. probably need to somehow restart the service for new rules to be tracked then do a reload_config to reload them?
 
> Can you just not provide the pidfile option at all?

This also confirms my intention :
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=223052
with a slightly different approach:
https://github.com/freebsd/freebsd-ports/blob/master/security/suricata/files/suricata.in#L61